File: src/main/java/no/nav/pensjon/selvbetjeningopptjening/security/impersonal/ClientCredentials.java
package no.nav.pensjon.selvbetjeningopptjening.security.impersonal;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ClientCredentials {

    private final String clientId;
    private final String clientSecret;

    public ClientCredentials(@Value("${azure-app.client-id}") String clientId,
                             @Value("${azure-app.client-secret}") String clientSecret) {
        this.clientId = clientId;
        this.clientSecret = clientSecret;
    }

    public String getClientId() {
        return clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }
}

File: src/main/java/no/nav/pensjon/selvbetjeningopptjening/tech/security/egress/oauth2/clientcred/ClientCredentials.kt
package no.nav.pensjon.selvbetjeningopptjening.tech.security.egress.oauth2.clientcred

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class ClientCredentials(
    @param:Value("\${azure-app.client-id}") val clientId: String,
    @param:Value("\${azure-app.client-secret}") val clientSecret: String
)
